### PR TITLE
feat(lookup): add lookup request rate limiter

### PR DESCRIPTION
**Pull request by mathpaquette**
#90

### CODE REVIEW

**Review by mathpaquette — COMMENTED** [review]
@Nucs @BunkerCoder could you please take some time to test it on your side?
https://www.myget.org/feed/iqfeedcsharpapiclient/package/nuget/IQFeed.CSharpApiClient/2.5.1.494

File: src/IQFeed.CSharpApiClient/Lookup/LookupRateLimiter.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+        private async void ReleaseTask(int intervalMs, int maxCount)
+        {
+            var intervalTicks = (int)TimeSpan.FromMilliseconds(intervalMs).Ticks;
+            var remainderTicks = 0;
+            var sw = new Stopwatch();
+
+            while (true)
+            {
+                sw.Restart();
```
[the remark this answers is not in the document]
Comment (BunkerCoder) [reply src/IQFeed.CSharpApiClient/Lookup/LookupRateLimiter.cs:34]: @mathpaquette - Awesome. Thanks. Still spinning up on git command line and fighting with it. Hope to be ready to test shortly.

[the remark this answers is not in the document]
Comment (BunkerCoder) [reply src/IQFeed.CSharpApiClient/Lookup/LookupRateLimiter.cs:34]: Installed 2.5.1.494 from myget. Initial tests looking good so far.

**Comment by BunkerCoder** [comment]
@mathpaquette  - I have been trying to keep up with your commits and keep testing.  I hijacked ConcurrentHistoricalExample because my failing client is modeled after it.  Unless I made mistakes, I'm seeing performance continuing to increase when setting RequestsPerSecond to values > 50.  Performance does contract below 50.  Checkout https://github.com/BunkerCoder/IQFeed.CSharpApiClient/tree/ratelimiter-tests and experiment with the test parameters.  I'll continue to look for mistakes, but wanted to flag this early.

**Comment by mathpaquette** [comment]
> @mathpaquette - I have been trying to keep up with your commits and keep testing. I hijacked ConcurrentHistoricalExample because my failing client is modeled after it. Unless I made mistakes, I'm seeing performance continuing to increase when setting RequestsPerSecond to values > 50. Performance does contract below 50. Checkout https://github.com/BunkerCoder/IQFeed.CSharpApiClient/tree/ratelimiter-tests and experiment with the test parameters. I'll continue to look for mistakes, but wanted to flag this early.

@BunkerCoder yes.. I know, this is because IQFeed turned off the rate limit for now so you still can go over 50+ requests per second but this wont be the case after January 12th if I remember. Just completed my tests on my end. Mostly ready to merge. Ill wait 1 or 2 days if someone notice anything critical. Thanks.

**Comment by BunkerCoder** [comment]
@mathpaquette Sorry...not thinking it through all the way before opening my mouth = MISTAKE.  It would be nice to have a test server...  I'll install 2.6.0.509 and continue testing my client.

**Comment by mathpaquette** [comment]
> @mathpaquette Sorry...not thinking it through all the way before opening my mouth = MISTAKE.  It would be nice to have a test server...  I'll install 2.6.0.509 and continue testing my client.

You can look at my unit test too. It makes it easier to understand how does that work by changing values. The tolerance is very small so it means that we gonna stay close as possible to the limit. I feel that my solution is pretty clever by keeping the modulo of ticks as reminder for the next iteration :)

**Comment by BunkerCoder** [comment]
> 
> 
> > @mathpaquette Sorry...not thinking it through all the way before opening my mouth = MISTAKE.  It would be nice to have a test server...  I'll install 2.6.0.509 and continue testing my client.
> 
> You can look at my unit test too. It makes it easier to understand how does that work by changing values. The tolerance is very small so it means that we gonna stay close as possible to the limit. I feel that my solution is pretty clever by keeping the modulo of ticks as reminder for the next iteration :)

Yes - Super-slick/quick.  Have not checked the unit tests yet, but will do.

**Pull request merged**